projects/common-form-elements/src/lib/keywords/keywords.component.ts

```diff
@@ -1,35 +1,100 @@
-import { Component, Input, OnInit } from '@angular/core';
-import { FormControl } from '@angular/forms';
-import { FieldConfig } from '../common-form-config';
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { CustomFormControl, DynamicFieldConfigOptionsBuilder, FieldConfig, FieldConfigOption } from '../common-form-config';
 import * as _ from 'lodash-es';
+import { merge, Observable, Subject, Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
 
 @Component({
   selector: 'sb-keywords',
   templateUrl: './keywords.component.html',
   styleUrls: ['./keywords.component.css']
 })
-export class KeywordsComponent implements OnInit {
+export class KeywordsComponent implements OnInit,OnChanges,OnDestroy {
   @Input() label: String;
   @Input() placeholder: String;
-  @Input() formControlRef: FormControl;
+  @Input() formControlRef: CustomFormControl;
   @Input() field: FieldConfig<String>;
   @Input() validations?: any;
   @Input() disabled: Boolean;
   @Input() default: String;
-
-
+  @Input() options: any;
+  @Input() formGroup?: FormGroup;
+  @Input() dataLoadStatusDelegate: Subject<'LOADING' | 'LOADED'>;
+  @Input() depends?: any;
   public items: any;
   inputText = '';
   selectedItems:any;
+  options$?: Observable<FieldConfigOption<any>[]>;
+  contextValueChangesSubscription?: Subscription;
+  latestParentValue: string;
   constructor() { }
 
+  ngOnChanges(changes: SimpleChanges): void {
+    console.log(this.options);
+   
+  }
+
   ngOnInit() {
     if (!_.isEmpty(this.default)) {
       this.items = this.default;
     }
     if(!_.isEmpty(this.field?.default)){
       this.selectedItems = this.field?.default;
     }
+
+    if (!this.options) {
+      this.options = _.isEmpty(this.field.options) ? this.isOptionsClosure(this.field.options) && this.field.options : [];
+    }
+
+    if (!_.isEmpty(this.depends) && !this.isOptionsClosure(this.options)) {
+      this.contextValueChangesSubscription =  merge(..._.map(this.depends, depend => depend.valueChanges)).pipe(
+       tap((value: any) => {
+         this.latestParentValue = value;
+       })
+       ).subscribe();
+     }
```

Comment:
are we making use of `this.latestParentValue` value anywhere? if not please remove this entire if condition for now and add the `this.contextValueChangesSubscription` to line.no 59.